wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -0,0 +1,1614 @@
+
+        for (String branch : branches) {
+            String r = config.getString(ConfigConstants.CONFIG_BRANCH_SECTION, branch,
+                                        ConfigConstants.CONFIG_KEY_REMOTE);
+            if (name.equals(r)) {
+                config.unset(ConfigConstants.CONFIG_BRANCH_SECTION, branch, ConfigConstants.CONFIG_KEY_REMOTE);
+                config.unset(ConfigConstants.CONFIG_BRANCH_SECTION, branch, ConfigConstants.CONFIG_KEY_MERGE);
+                List<Branch> remoteBranches = branchList(newDto(BranchListRequest.class).withListMode("r"));
+                for (Branch remoteBranch : remoteBranches) {
+                    if (remoteBranch.getDisplayName().startsWith(name)) {
+                        branchDelete(newDto(BranchDeleteRequest.class).withName(remoteBranch.getName()).withForce(true));
+                    }
+                }
+            }
+        }
+
+        try {
+            config.save();
+        } catch (IOException exception) {
+            throw new GitException(exception.getMessage(), exception);
+        }
+    }
+
+    @Override
+    public List<Remote> remoteList(RemoteListRequest request) throws GitException {
+        StoredConfig config = repository.getConfig();
+        Set<String> remoteNames = new HashSet<>(config.getSubsections(ConfigConstants.CONFIG_KEY_REMOTE));
+        String remote = request.getRemote();
+
+        if (remote != null && remoteNames.contains(remote)) {
+            remoteNames.clear();
+            remoteNames.add(remote);
+        }
+
+        List<Remote> result = new ArrayList<>(remoteNames.size());
+        for (String remoteName : remoteNames) {
+            try {
+                List<URIish> uris = new RemoteConfig(config, remoteName).getURIs();
+                result.add(newDto(Remote.class).withName(remoteName).withUrl(uris.isEmpty() ? null : uris.get(0).toString()));
+            } catch (URISyntaxException exception) {
+                throw new GitException(exception.getMessage(), exception);
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public void remoteUpdate(RemoteUpdateRequest request) throws GitException {
+        String remoteName = request.getName();
+        if (remoteName == null || remoteName.length() == 0) {
+            throw new IllegalArgumentException(ERROR_UPDATE_REMOTE_NAME_MISSING);
+        }
+
+        StoredConfig config = repository.getConfig();
+        Set<String> remoteNames = config.getSubsections(ConfigConstants.CONFIG_KEY_REMOTE);
+        if (!remoteNames.contains(remoteName)) {
+            throw new IllegalArgumentException("Remote " + remoteName + " not found. ");
+        }
+
+        RemoteConfig remoteConfig;
+        try {
+            remoteConfig = new RemoteConfig(config, remoteName);
+        } catch (URISyntaxException e) {
+            throw new GitException(e.getMessage(), e);
+        }
+
+        List<String> tmp;
+
+        tmp = request.getBranches();
```

Comment:
* bad name for variable.
* We can init variable near declatation

@vinokurig also the same to https://github.com/eclipse/che/pull/1297/files#r63862165 here. Please check whole pull request that collections from DTOs are not checked to null values